ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/keyboard.html

```diff
@@ -49,8 +82,8 @@
             }
             
             function setOskHeight(height) {
-                oskHeight = height;
                 var kmw=window['keyman'];
+                oskHeight = height - kmw.osk.banner.height;
```

Comment:
hmm, on Android `oskHeight = height` and doesn't use the banner height...

Reply:
So, how do you preserve the banner's space in the Android app?

Reply:
In the Android app, `oskHeight` is initialized.
But all the other calls `setOskHeight(0)` don't actually change the heights:
```
    function setOskHeight(h) {
      if(h > 0) {
        oskHeight = h;
      }
      var kmw=window['keyman'];
      kmw['correctOSKTextSize']();
    }
```

Banner height does get updated as the banner toggles between "blank" and "suggestion".

And the entire WebView height gets adjusted accordingly.

Reply:
Yeah, iOS works pretty differently there.  iOS tries to default to the system keyboard's preferred height (via constraints), so its height changes with rotations, etc.  As a result, those calls will directly change the amount of vertical screen space the OSK gets - and so iOS needs to ensure that it's changing the _keyboard's_ space and leaving appropriate room for the banner on each update.

In case it's not otherwise clear from that, the `WKWebView` manages KMW's total height.  With the implementation in this PR, the main before-and-after difference is that the in-app keyboard will be squeezed down when predictive text is active.  (There's no effect on the system keyboard, as it's always used a banner.)